spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/alloydb/AlloyDbProperties.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.autoconfigure.alloydb;
+
+import com.google.cloud.spring.core.Credentials;
+import com.google.cloud.spring.core.CredentialsSupplier;
+import com.google.cloud.spring.core.GcpScope;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.NestedConfigurationProperty;
+
+@ConfigurationProperties("spring.cloud.gcp.alloydb")
+public class AlloyDbProperties implements CredentialsSupplier {
```

Comment:
Can we cross-reference this with the properties supported for Cloud SQL to make sure that we have parity in the integrations to the extent where it makes sense?

Reply:
`AlloyDbProperties` contains all configurable properties of the [AlloyDB Java Connector](https://github.com/GoogleCloudPlatform/alloydb-java-connector/blob/main/docs/jdbc.md). 

Most of them match the `GcpCloudSqlProperties`.

Please let me know if there is anything missing.